releases/release-4.0.10.md

```diff
@@ -0,0 +1,68 @@
+---
+title: TiDB 4.0.10 Release Notes
+---
+
+# TiDB 4.0.10 Release Notes
+
+Release date: January 15, 2021
+
+TiDB version: 4.0.10
+
+## Compatibility Changes
+
+## New Features
+
+## Improvements
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix a concurrency bug that may cause the batch client timeout [#22336](https://github.com/pingcap/tidb/pull/22336)
+    - Avoid duplicate bindings caused by concurrent baseline capture [#22295](https://github.com/pingcap/tidb/pull/22295)
+    - Make baseline capture work when log level is 'debug' [#22293](https://github.com/pingcap/tidb/pull/22293)
+    - Fix gc miss locks when region merged during scanning & resolving locks [#22267](https://github.com/pingcap/tidb/pull/22267)
+    - Return correct results for user variables of datetime type [#22143](https://github.com/pingcap/tidb/pull/22143)
+    - Avoid using index merge when there are multiple table filters. [#22124](https://github.com/pingcap/tidb/pull/22124)
+    - Fixed the issue that using prepare statement to cache execution plan caused tiflash to return `Wrong precision`. [#21960](https://github.com/pingcap/tidb/pull/21960)
+    - Fix a bug that schema change will lead to incorrectness when forUpdateRead [#21596](https://github.com/pingcap/tidb/pull/21596)
+    - Fix a bug that doesn't add NoDefaultValueFlag when creating a table [#21474](https://github.com/pingcap/tidb/pull/21474)
+    - Set dbName for hinted query block table alias [#21380](https://github.com/pingcap/tidb/pull/21380)
+    - Generate proper hint for IndexHashJoin / IndexMergeJoin [#21020](https://github.com/pingcap/tidb/pull/21020)
+
++ TiFlash
+
+    - Fix the problem that TiFlash fail to start when path_realtime_mode is true [#1322](https://github.com/pingcap/tics/pull/1322)
+    - Fixed an issue where tables created in the old version of tidb may cause the startup of the new tiflash node to fail. [#1297](https://github.com/pingcap/tics/pull/1297)
+
+## 未分类 notes。请对以下 Note 进行分类，挪动到上面 Compatibility changes、New features、Improvement 分类下
+
++ TiDB
+
+    - Session: fix the duplicate binding case when updating bind info [#22368](https://github.com/pingcap/tidb/pull/22368)
+    - Session: fix two cases when updating bind info [#22341](https://github.com/pingcap/tidb/pull/22341)
+    - Make txn entry size limit configurable [#21843](https://github.com/pingcap/tidb/pull/21843)
+
++ TiKV
+
+    - Fix wrong mapping between readies and peers by offset [#9409](https://github.com/tikv/tikv/pull/9409)
+    - Fix some logs not being redact when `security.redact-info-log = true` [#9314](https://github.com/tikv/tikv/pull/9314)
+
++ PD
```

Comment:
some changes:
# Bug Fixes
- Fix the id allocation is not monotonic [#3308](https://github.com/tikv/pd/pull/3308) [#3323](https://github.com/tikv/pd/pull/3323)
- Fix the pd client could be blocked in some cases [#3285](https://github.com/pingcap/pd/pull/3285)
# New features
- Support configuration `enable-redact-log` to enable log desensitization or not. [#3266](https://github.com/pingcap/pd/pull/3266)
# Improvement
- Optimize the store state filter metrics [#3100](https://github.com/tikv/pd/pull/3100)
- Upgrade go.etcd.io/bbolt to v1.3.5 [#3331](https://github.com/tikv/pd/pull/3331)